net/http/authn/context.go

```diff
@@ -1,14 +1,32 @@
 package authn
 
-import "context"
+import (
+	"context"
+	"crypto/x509/pkix"
+)
 
 type key int
 
 const (
 	tokenKey key = iota
 	localhostKey
+	subjectNameKey
 )
 
+// newContextWithSubjectName sets the cert subject name in a new context and returns that context
+func newContextWithSubjectName(ctx context.Context, name *pkix.Name) context.Context {
+	return context.WithValue(ctx, subjectNameKey, name)
+}
+
+// SubjectName returns the cert subject name stored in the context, if it exists.
+func SubjectName(ctx context.Context) *pkix.Name {
```

Comment:
I think the name for this should be `X509SubjectName` or `X509Certs` (if the cert list is in the context).